src/main/resources/public/ts/models/badge-protagonist-setting.model.ts

```diff
@@ -0,0 +1,53 @@
+import {MinibadgeModel} from "./model";
+
+
+export interface IBadgeProtagonistSettingResponse {
+    type:string;
+    typeId :string;
+}
+
+
+export interface IBadgeProtagonistSettingRelationResponse {
+    assignorType:BadgeProtagonistSetting;
+    receiverType :BadgeProtagonistSetting;
+}
+
+export class BadgeProtagonistSettingRelation extends MinibadgeModel<BadgeProtagonistSettingRelation> {
+    assignorType:BadgeProtagonistSetting;
+    receiverType :BadgeProtagonistSetting;
+
+    constructor(data?: IBadgeProtagonistSettingRelationResponse) {
+        super();
+        if (data) this.build(data);
+    }
+
+    build(data: IBadgeProtagonistSettingRelationResponse): BadgeProtagonistSettingRelation {
+        this.receiverType = data.receiverType;
+        this.assignorType = data.assignorType;
```

Comment:
Dans la logique des choses ici, ton data ne devrait être constitué seulement de différents éléments de réponses imbriqués:
IBadgeProtagonistSettingRelationResponse => IBadgeProtagonistSettingResponse.

du coup, il va falloir ici que tu instancie chaque morceau en un nouveau model, exemple:
```suggestion
        this.assignorType = new BadgeProtagonistSetting(<IBadgeProtagonistSettingResponse>data.assignorType)
```